### ISSUE TITLE
[aws-eks] EKS FargateCluster fails to create with "r\nerror: the server doesn\'t have a resource type "deployment"\n'"

### ISSUE THREAD

**Issue by fmcmac** [aws/aws-cdk#8857]
Attempt to create an EKS cluster with minimal python code:

```
# create the eks cluster
eks_cluster = eks.FargateCluster(
    self,
    "eks_cluster",
    vpc=vpc,
    vpc_subnets=vpc.private_subnets,
)
```
returns an error as per error log below.

### Reproduction Steps

To reproduce use the python code above and run ```cdk deploy```

### Error Log

17:36:28 | CREATE_FAILED        | Custom::AWSCDK-EKS-KubernetesPatch    | eks_cluster/CoreDn...h/Resource/Default
Failed to create resource. Error: b'E0702 05:35:35.532880      13 round_trippers.go:174] CancelRequest not implemented by *exec.roundTrippe
r\nerror: the server doesn\'t have a resource type "deployment"\n'
at invokeUserFunction (/var/task/framework.js:85:19)
at process._tickCallback (internal/process/next_tick.js:68:7) 

### Environment

  - **CLI Version:**  1.47.0 (build c2b499a)
  - **Framework Version:**
  - **Node.js Version:** v12.13.0
  - **OS: MacOs
  - **Language (Version):** python 3.8.1

### Other

--- 

This is :bug: Bug Report

**Comment by eladb** [comment]
There is a known issue with Fargate which is being addressed as part of #8859 and will be released next week. I believe this is related.

**Comment by eladb** [comment]
Please reopen if this persists after the next CDK upgrade (coming this week).
